include/boost/geometry/algorithms/detail/disjoint/segment_box.hpp

```diff
@@ -251,6 +251,110 @@ struct disjoint_segment_box
     }
 };
 
+
+struct disjoint_segment_box_with_info_compute
+{
+    template <typename Segment, typename Box, typename Strategy, typename Result>
+    static inline void apply(Segment const& segment,
+                             Box const& box,
+                             Strategy const& strategy,
+                             Result& result)
+    {
+        assert_dimension_equal<Segment, Box>();
+
+        typedef typename geometry::point_type<Segment>::type segment_point;
+
+        segment_point p[2];
+        detail::assign_point_from_index<0>(segment, p[0]);
+        detail::assign_point_from_index<1>(segment, p[1]);
+
+        if (geometry::covered_by(p[0], box,
+                                 strategy.get_disjoint_point_box_strategy()))
+        {
+            result.count = 1;
+            result.intersections[0] = p[0];
+            return;
+        }
+        if (geometry::covered_by(p[1], box,
+                                 strategy.get_disjoint_point_box_strategy()))
+        {
+            result.count = 1;
+            result.intersections[0] = p[1];
+            return;
+        }
+
+        typedef typename geometry::point_type<Box>::type box_point;
+        box_point top_left, top_right, bottom_left, bottom_right;
+        detail::assign_box_corners(box, bottom_left, bottom_right,
+                                   top_left, top_right);
+
+        // check segment intersection with box meridians
+        // which are (geodesic) segments on all coordinate systems
+        geometry::model::segment<segment_point> s
+                = geometry::model::segment<segment_point>(p[0], p[1]);
+        strategy.get_disjoint_segment_box_with_info_strategy()
+                .apply(s, bottom_left, top_left,
+                       strategy.get_relate_segment_segment_strategy(),
+                       result);
```

Comment:
This is probably not the way to go. A strategy taking another strategy? If needed `disjoint_segment_box_with_info_strategy` should simply contain the correct `relate_segment_segment_strategy`. Or is there a problem with cyclic dependency? But then I wonder why would `relate_segment_segment_strategy` need any `_with_info` strategy?

Reply:
Indeed there is a problem with cyclic dependency. Here, `strategy` can be either a closest_point, distance, disjoint or intersection strategy. All of them provide the disjoint_segment_box_with_info_strategy and relate_segment_segment_strategy.